sheetshuttle/util.py

```diff
@@ -26,6 +26,23 @@ def gh_token_exists() -> bool:
     return False
 
 
+def extract_sheet_id(url: str) -> str:
+    """Extract a sheet ID from a Google Sheets URL."""
+    try:
+        url_segments = url.split("/")
+        return next(
+            (
+                url_segments[i + 1]
+                for i in range(len(url_segments))
+                if url_segments[i] == "d"
+            ),
+            url,
+        )
+    except IndexError:
+        print("The URL provided is not valid for Google Sheets ID extraction.")
+        return url
```

Comment:
Can we print the URL here as well. It might help the user debug the issue they have. But most importantly, why are we still returning the url when we know it's not valid? Shouldn't we raise the exception with our error message?
I think it should look something like this:
```suggestion
    except IndexError:
        raise Exception(f"The URL: {url} is not valid for Google Sheets ID extraction.")

```

Reply:
One of the lint checks was having an issue with our `except` block not having a return statement. I'm not sure of a way around it without returning *something*, but wouldn't know what to return if not the original url. Thoughts?